server: TestCachedSettingsServerRestart failed

server.TestCachedSettingsServerRestart [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12046147?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12046147?buildTab=artifacts#/) on release-23.1 @ [1bf58a144bf53b0480cdd108d269fc7724875cf1](https://github.com/cockroachdb/cockroach/commits/1bf58a144bf53b0480cdd108d269fc7724875cf1):


```
=== RUN   TestCachedSettingsServerRestart
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestCachedSettingsServerRestart3328014851
    test_log_scope.go:79: use -show-logs to present logs inline
    settings_cache_test.go:128: condition failed to evaluate within 3m45s: initial state settings KVs does not match expected settings
        Expected: [{Key:/Table/6/1/"diagnostics.reporting.enabled"/0 Value:{RawBytes:[82 106 93 86 10 38 4 116 114 117 101 24 166 227 234 209 12 240 212 196 228 4 22 1 98] Timestamp:0,0}}]
        Actual:   [{Key:/Table/6/1/"diagnostics.reporting.enabled"/0 Value:{RawBytes:[82 106 93 86 10 38 4 116 114 117 101 24 166 227 234 209 12 240 212 196 228 4 22 1 98] Timestamp:0,0}} {Key:/Table/6/1/"version"/0 Value:{RawBytes:[229 169 129 172 10 38 10 18 8 8 23 16 1 24 0 32 0 24 166 227 234 209 12 208 148 211 179 6 22 1 109] Timestamp:0,0}}]
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b2c9b3a394428c7572d34050aad8975/logTestCachedSettingsServerRestart3328014851
--- FAIL: TestCachedSettingsServerRestart (230.42s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCachedSettingsServerRestart.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/settings_cache_test.go
+++ b/pkg/server/settings_cache_test.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/storage"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
@@ -79,8 +80,21 @@ func TestCachedSettingsServerRestart(t *testing.T) {
 	}
 	var settingsCache []roachpb.KeyValue
 	testServer, _, _ := serverutils.StartServer(t, serverArgs)
-	closedts.TargetDuration.Override(ctx, &testServer.ClusterSettings().SV, 10*time.Millisecond)
-	closedts.SideTransportCloseInterval.Override(ctx, &testServer.ClusterSettings().SV, 10*time.Millisecond)
+
+	factor := 1
+	if util.RaceEnabled {
+		// Under race, all the goroutines are generally slower. If we
+		// accelerate the rangefeeds and the closed ts framework too much,
+		// it can start overwhelming the scheduler and starve everything
+		// of CPU time. So give everything some breathing time in that
+		// case.
+		//
+		// TODO(knz): Replace this by the change in #111753.
+		factor = 4
+	}
+	closedts.TargetDuration.Override(ctx, &testServer.ClusterSettings().SV, 10*time.Millisecond*time.Duration(factor))
+	closedts.SideTransportCloseInterval.Override(ctx, &testServer.ClusterSettings().SV, 10*time.Millisecond*time.Duration(factor))
+
 	testutils.SucceedsSoon(t, func() error {
 		store, err := testServer.GetStores().(*kvserver.Stores).GetStore(1)
 		if err != nil {

